src/legacy/core_plugins/kibana/public/visualize/wizard/type_selection/type_selection.tsx

```diff
@@ -44,9 +46,14 @@ interface VisTypeListEntry extends VisType {
   highlighted: boolean;
 }
 
+interface VisTypeAliasListEntry extends VisTypeAlias {
+  highlighted: boolean;
+}
+
 interface TypeSelectionProps {
   onVisTypeSelected: (visType: VisType) => void;
   visTypesRegistry: VisType[];
+  visTypeAliases: VisTypeAlias[];
   showExperimental: boolean;
```

Comment:
Essentially, we need another switch like this one `showExperimental` that is something like `showExternal` which will add another case to the `stage`: https://github.com/elastic/kibana/blob/11e503538d73200c77f87ce00f6cd2522f33a89a/src/legacy/core_plugins/kibana/public/visualize/wizard/type_selection/type_selection.tsx#L201-L212
That adds a "betaBadge" with some options similar to:

```
betaBadgeLabel="External application"
betaBadgeTooltipContent="This item will lead to a separate application within Kibana, but outside of Visualize."
betaBadgeIconType="popout"
```

Though after using the word 'external' a bunch, I don't think that's appropriate as it could be confusing that it would take you outside of Kibana. So hopefully @gchaps can help there and with the rest of the copy.

--

Then the same content should be added to the description on the right side of the modal.